### PR TITLE
Refactor StorageProviderBase.on() and .off()

### PR DESCRIPTION
**Pull request by csilvestrini**
These methods now have the same API as the new storage stack's on and
off methods. The old behaviour is preserved in legacyOn and legacyOff
methods (there's a bunch of tests that rely on that old behaviour, but
they are all specific to the old storage stack and so will be deleted
later).

### CODE REVIEW

File: src/planning/plan/plan-producer.ts [hunk 1]
```diff
@@ -123,11 +125,12 @@ export class PlanProducer {
 
   dispose() {
     if (this.searchStore) {
-      this.searchStore.off(this.searchStoreCallback);
+      this.searchStore.off(this.searchStoreCallbackId);
     }
   }
 
   async produceSuggestions(options: SuggestionOptions = {}) {
+    console.log(`produceSuggestions()`);
```
Comment (Cypher1) [inline src/planning/plan/plan-producer.ts]: Possibly unintended

File: src/runtime/arc.ts [hunk 2]
```diff
@@ -738,7 +738,10 @@ ${this.activeRecipe.toString()}`;
     this.storeTags.set(store, new Set(tags));
 
     this.storageKeys[store.id] = store.storageKey;
-    store.on(() => this._onDataChange());
+    store.on(async () => {
```
Comment (Cypher1) [inline src/runtime/arc.ts]: Might be simpler to have onDataChange return true? (I've seen a couple of these).

Comment (csilvestrini) [reply src/runtime/arc.ts]: Done

File: src/planning/plan/plan-producer.ts (hunk 1 above)
Comment (csilvestrini) [reply src/planning/plan/plan-producer.ts]: Thanks, done

**Pull request merged**